pkg/deployment/depresource/new_deployment.go

```diff
@@ -24,46 +24,45 @@ import (
 	"github.com/elastic/cloud-sdk-go/pkg/models"
 )
 
+// InstanceParams holds the common instance fields for a resource
+type InstanceParams struct {
+	Size      int32
+	ZoneCount int32
+	RefID     string
+}
+
 // NewParams is consumed by New()
 type NewParams struct {
 	*api.API
 
-	Name                   string
-	Version                string
-	DeploymentTemplateID   string
-	Region                 string
-	ElasticsearchRefID     string
-	KibanaRefID            string
-	ApmRefID               string
-	AppsearchRefID         string
-	ElasticsearchSize      int32
-	ElasticsearchZoneCount int32
-	KibanaSize             int32
-	KibanaZoneCount        int32
-	ApmSize                int32
-	ApmZoneCount           int32
-	AppsearchSize          int32
-	AppsearchZoneCount     int32
-	ApmEnable              bool
-	AppsearchEnable        bool
-	Writer                 io.Writer
-	Plugins                []string
-	TopologyElements       []string
+	Name                  string
+	Version               string
+	DeploymentTemplateID  string
+	Region                string
+	ApmEnable             bool
+	AppsearchEnable       bool
+	Writer                io.Writer
+	Plugins               []string
+	TopologyElements      []string
+	ElasticsearchInstance InstanceParams
```

Comment:
Same comment here about the fields from ElasticsearchInstance to ElasticsearchResource